### ISSUE TITLE
text() returns only first element's text nodes, contra JQuery

### ISSUE THREAD

**Issue by cefn** [madrobby/zepto#1142]
I ran into a problem porting some JQuery-based in-browser testing from JQuery to Zepto, worth commenting but I recognise JQuery equivalence is a bonus not a guarantee.

Turned out that the contract of text() in JQuery and Zepto are distinctly different, with http://api.jquery.com/text/ indicating that JQuery#text() called on a set of selected elements should return the combined text nodes of all selected items, and Zepto#text() promising that it returns only the text nodes of the first of the selected items. Looks like .html has a similar issue. This may easily be overlooked in a test suite where the selector only matches a single item.

Not sure what should be done about this, but couldn't identify an issue discussing this bug/feature, so this might be a useful place to record the design decision if it's deliberate, or track the change if this variation wasn't intended.

It wasn't super-easy for me to solve as the $('selector') text() call was injected by the test suite through Webdriver, and required a much more complicated injection to do traversal and normalise between JQuery and Zepto. As a workaround...

```javascript
$(selector).text()
```

...becomes...
```javascript
Array.prototype.slice.apply($(selector)).map(function(item){ return $(item).text(); }).join("")
```

P.S. One of the complexities I experienced was that Zepto's array-like behaviour was unexpected, with join() missing and map() having inverted argument types relative to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map (in other words the map function needed the non-standard signature function(index, item){} instead of function(item){} ).

**Comment by mislav** [comment]
You're right about `fn.text()`. We should be returning all text items concatenated into one string without a space delimiter, instead of just returning the first item. Would you create a pull request to fix this?

In the meantime, you could use:

```js
$(selector).map(function(){ return this.textContent }).get().join("")
```

which is a tiny bit shorter than your workaround and doesn't need `Array.prototype.slice`.

As for `fn.join()` missing, jQuery doesn't seem to have it either, so we never implemented it. As for `fn.map()` having reversed arguments in contrast to `Array.prototype.map`, that's by design and for jQuery compatibility. jQuery expects `fn.map()` function like this: `Function(Integer index, Element domElement)`. So I don't believe we're doing anything wrong here.

**Comment by mislav** [comment]
Fixed by #1152
